stac_generator/typing.py

```diff
@@ -0,0 +1,98 @@
+import datetime
+from typing import Literal, TypedDict
+
+import pandas as pd
+
+DTYPE = Literal[
+    "str",
+    "int",
+    "bool",
+    "float",
+    "int8",
+    "int16",
+    "int32",
+    "int64",
+    "uint8",
+    "uint16",
+    "uint32",
+    "uint64",
+    "float16",
+    "float32",
+    "float64",
+    "cint16",
+    "cint32",
+    "cfloat32",
+    "cfloat64",
+    "other",
+]
+
+# Generic Types
+
+NumberT = int | float
+DateTimeT = datetime.datetime
+BBoxT = list[NumberT]  # Either 2D or 3D BBoxType
+TimeExtentT = tuple[DateTimeT | None, DateTimeT | None]  # (start_datetime, end_datetime) format
+CSVMediaType = "text/csv"  # https://www.rfc-editor.org/rfc/rfc7111
+ExcelMediaType = "application/vnd.ms-excel"  # https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+FrameT = pd.DataFrame
+
+# GeoJSON types
+
+GeometryT = Literal[
```

Comment:
Is it possible to use an existing definition for geometry types, such as [shapely geometries](https://shapely.readthedocs.io/en/stable/geometry.html)?

Reply:
Addressed in the latest commit